chainstate/test-framework/src/random_tx_maker.rs

```diff
@@ -21,35 +21,82 @@ use chainstate::chainstate_interface::ChainstateInterface;
 use common::{
     chain::{
         output_value::OutputValue,
+        stakelock::StakePoolData,
+        timelock::OutputTimeLock,
         tokens::{
             make_token_id, IsTokenUnfreezable, NftIssuance, TokenId, TokenIssuance,
             TokenTotalSupply,
         },
-        AccountCommand, AccountNonce, AccountType, Destination, Transaction, TxInput, TxOutput,
-        UtxoOutPoint,
+        AccountCommand, AccountNonce, AccountOutPoint, AccountSpending, AccountType, DelegationId,
+        Destination, OutPointSourceId, PoolId, Transaction, TxInput, TxOutput, UtxoOutPoint,
     },
-    primitives::{Amount, BlockHeight},
+    primitives::{per_thousand::PerThousand, Amount, BlockHeight, Id, H256},
+};
+use crypto::{
+    random::{CryptoRng, Rng, SliceRandom},
+    vrf::{VRFKeyKind, VRFPrivateKey},
 };
-use crypto::random::{CryptoRng, Rng};
 use itertools::Itertools;
+use pos_accounting::{
+    make_pool_id, DelegationData, InMemoryPoSAccounting, PoSAccountingDB, PoSAccountingDelta,
+    PoSAccountingDeltaData, PoSAccountingOperations, PoSAccountingUndo, PoSAccountingView,
+    PoolData,
+};
 use test_utils::nft_utils::*;
 use tokens_accounting::{
     InMemoryTokensAccounting, TokensAccountingCache, TokensAccountingDB, TokensAccountingDeltaData,
     TokensAccountingOperations, TokensAccountingView,
 };
 use utxo::Utxo;
 
+// TODO: currently this returns random pool from store but only if it exists in delta. This gives None more often.
+//       There should be a way to return random result after delta is accounted.
+//       Same applies for delegation.
+fn get_random_pool_data<'a>(
+    rng: &mut impl Rng,
+    storage: &'a InMemoryPoSAccounting,
+    tip_view: &impl PoSAccountingView,
+) -> Option<(&'a PoolId, &'a PoolData)> {
+    let all_pool_data = storage.all_pool_data();
+    (!all_pool_data.is_empty())
+        .then(|| all_pool_data.iter().nth(rng.gen_range(0..all_pool_data.len())).unwrap())
+        .and_then(|(id, data)| tip_view.pool_exists(*id).unwrap().then_some((id, data)))
+}
+
+fn get_random_delegation_data<'a>(
+    rng: &mut impl Rng,
+    storage: &'a InMemoryPoSAccounting,
+    tip_view: &impl PoSAccountingView,
+) -> Option<(&'a DelegationId, &'a DelegationData)> {
+    let all_delegation_data = storage.all_delegation_data();
+    (!all_delegation_data.is_empty())
+        .then(|| {
+            {
+                all_delegation_data
+                    .iter()
+                    .nth(rng.gen_range(0..all_delegation_data.len()))
+                    .unwrap()
```

Comment:
The `crypto::random` module re-exports the `IteratorRandom` trait that can be used to sample from a iterator, or `SliceRandom` which can be used to sample from a slice. Would it make this more readable? (I believe I've seen multiple occurrences of this pattern).